background.js

```diff
@@ -59,46 +50,22 @@ function is_hostname_blocked(url, tabId) {
   return false;
 }
 
-chrome.tabs.onCreated.addListener(function (tab) {
-  // console.log(tab.id)
-  // chrome.tabs.sendMessage(tab.id, "hello world");
-});
-
 chrome.tabs.onUpdated.addListener((tabId, changeInfo, tab) => {
   if (changeInfo && changeInfo.status && changeInfo.status == "complete") {
     // check if this is one of the blocked url
 
     if (is_hostname_blocked(tab.url, tab.id)) {
       console.log(tabId, changeInfo, tab, "sending overlay html");
       // here passing the overlay html, don't think it's the best practice
-      fetch(chrome.runtime.getURL("overlay/overlay.html"))
-        .then((response) => response.text())
-        .then((data) => chrome.tabs.sendMessage(tab.id, data));
+
+      chrome.tabs.executeScript(tab.id, {
+        file: "overlay/overlay.js",
+      });
       return;
     }
   }
 });
 
-chrome.webRequest.onBeforeRequest.addListener(
```

Comment:
why dont we want to block requests anymore? e.g. videos can auto play in the background?

Reply:
I thought you just wanted to visually block the page. In that case we could also mute the tab, lol. 
This PR does not block the request anymore because its implementation will not reload the tab anymore, so it cannot block the request, otherwise page will stop loading even after the user clicked the buttons. Maybe I need to rewrite the script injection and blockage logics.

Reply:
why was necessary to remove the reloading initially? is it because the content script is injected differently?

Reply:
this part has two designs in mind:

1. when the page first loads, we let only the first HTML render, otherwise, chrome will show an error page and will not be able to inject the overlay
2. when the timer expires, it will start blocking xhr again, so activities will be paused again

a downside of this approach is that a reload is required after the overlay is dismissed

Reply:
That's why I prefer a visual block instead of the actual network block. In my opinion, blocking network is too much for this plugin. It's also unnecessary in order to implement the blockage. But I will split this pull request and keep its original injection logic.

Reply:
1. can you elaborate on why it's "too much" for this plugin?
2. the main purpose was to block auto-playing, e.g. youtube won't start buffering, I guess the debate is that if we want to block the webpage's activity when the overlay is visible?

Reply:
1. IMO, it's more about the philosophy of mindfulness, the emotional perspective, not the technical perspective. Just like the wallpaper with those three sentences, it's acting like a friend (or maybe the conscious of oneself) kindly reminds you to become mindful of what you are going to do. While blocking the network requests is like an enforcement, it might trigger one's negative emotion, which I think is the opposite of what this plugin wants to achieve.
2. I agreed with your concerns of auto-playing, but I think technically, we could pause the activity of a tab more gracefully  than blocking it. At least there are some plugins to stop auto-playing that we could refer to.